.github/workflows/commit-checker.yml

```diff
@@ -0,0 +1,42 @@
+name: Commit checker
+
+on:
+  pull_request:
+
+jobs:
+  commit-checker:
+    name: Commit checker
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout ${{ github.base_ref }} branch
+      uses: actions/checkout@v2
+      with:
+        ref: ${{ github.base_ref }}
+
+    - name: Switch to ${{ github.ref }} branch
+      run: |
+        git -c protocol.version=2 fetch --no-tags --prune --negotiation-tip=origin/${GITHUB_BASE_REF} origin +${GITHUB_SHA}:${GITHUB_REF}
```

Comment:
please enable 'set -x', as it is impossible to see what it really did like this :D

I have to experiment a bit to see if this really does what you hope it does in several scenarios .. hard to estimate when looking at it like this

Reply:
Okay, this is a bit difficult to test what it does exactly etc. It seems to only fetch the (full) history of 'master', and not the release branches etc. So there is some speed-up. Of course ideal we want to only fetch the commits between master and pull-request, but that seems to be a difficult thing to do in git-world.

I do have to say that I find this to have a lot of complexity, of which I am not sure it always works. On the other hand I am tempted to make a pull-request in `actions/checkout`, adding the `negotiation-tip`. Puzzling :)

Reply:
Okay, this is a bit difficult to test what it does exactly etc. It seems to only do what you hope it does when the pull-request is pretty recently rebased on master. Otherwise the full history is fetched, and it takes as long as just using the checkout action without comments.

Is that worth the complexity, I wonder?